demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1035 @@
+        ulCommandPayloadLength = prvInvokeMaxMinCommand( pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+
+        if( ulCommandPayloadLength > 0 )
+        {
+            if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 200,
+                                                       ucCommandPayloadBuffer,
+                                                       ulCommandPayloadLength ) != eAzureIoTSuccess )
+            {
+                LogInfo( ( "Error sending command response" ) );
+            }
+        }
+        else
+        {
+            LogInfo( ( "Error generating command payload" ) );
+        }
+    }
+    else
+    {
+        /* Not for max min report (not for this device) */
+        LogInfo( ( "Received command is not for this device" ) );
+
+        if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 404,
+                                                   sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                   strlen( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) ) != eAzureIoTSuccess )
+        {
+            LogInfo( ( "Error sending command response" ) );
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+static AzureIoTResult_t prvProcessProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                              AzureIoTHubClientPropertyType_t xPropertyType,
+                                              double * xOutTemperature,
+                                              uint32_t * ulOutVersion )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONTokenType_t xTokenType;
+    const uint8_t * pucComponentName = NULL;
+    uint32_t ulComponentNameLength = 0;
+
+    *xOutTemperature = 0.0;
+
+    xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_GetPropertiesVersion( &xAzureIoTHubClient, &xReader, pxMessage->xMessageType, ulOutVersion );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogInfo( ( "Error getting the property version" ) );
+    }
+    else
+    {
+        /* Reset JSON reader to the beginning */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xAzureIoTHubClient, &xReader,
+                                                                                 pxMessage->xMessageType, xPropertyType,
+                                                                                 &pucComponentName, &ulComponentNameLength ) ) == eAzureIoTSuccess )
+        {
+            xResult = AzureIoTJSONReader_TokenType( &xReader, &xTokenType );
+            configASSERT( xResult == eAzureIoTSuccess );
```

Comment:
this is agressing way of handling these error ? Parsing error should be reported back and possibly back to cloud with some error code.

Reply:
Yes and no. There's cases here where the hub may be sending malformed JSON in which case I don't think it's too crazy. Cases where wrong values might have been sent could maybe be handled